initialize_internal_tide.jl

```diff
@@ -199,60 +201,87 @@ Rig = RichardsonNumber(model; location=(Center, Center, Face), add_background=tr
 ε = KineticEnergyDissipationRate(model)
 χ = TracerVarianceDissipationRate(model, :b)/Bz
 
-slice_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
-point_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+# set the ouput mode:
+if output_mode == "spinup"
+        checkpoint_interval = 10days
+        slice_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+        point_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+        threeD_diags = (; ε, χ, uhat=û, what=ŵ, v=v,  B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+
+elseif output_mode == "test"
+        checkpoint_interval = tᶠ
+        slice_diags = (; ε, χ, uhat=û, B=B, b=b, Bz=Bz, uhat_z=uz)
+        threeD_diags = (; ε, χ, uhat=û, what=ŵ,  B=B, b=b, Bz=Bz, uhat_z=uz)
+
+else output_mode == "analysis"
+        checkpoint_interval = 10days
+        threeD_snapshot_interval=4Δtᵒ
+        slice_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+        point_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+        threeD_diags = (; ε, χ, uhat=û, what=ŵ, v=v, B=B, b=b, Bz=Bz, uhat_z=uz, Rig=Rig)
+
+end
 
 fname = string("internal_tide_theta=",string(θ),"_realtopo3D_Nx=",Nx,"_Nz=",Nz)
-dir = string("output/",simname)
+dir = string("output/",simname, "/")
 if output_writer
     ## checkpoint  
     simulation.output_writers[:checkpointer] = Checkpointer(
                                             model,
-                                            schedule=TimeInterval(5days),
+                                            schedule=TimeInterval(checkpoint_interval),
                                             dir=dir,
                                             prefix=string(fname, "_checkpoint"),
-                                            cleanup=false)
+                                            cleanup=true)
```

Comment:
Should cleanup also be a keyword argument? I don't remember what it does but it seems like we might want to be able to toggle it back and forth?

Reply:
Yeah, to my understanding, I use `cleanup=true`, in situations where either I start a new simulation or I want to discard the old runs. I use `cleanup=false` when I have picked up previous simulations. 
Same is true for `NetCDFOutputWriter(overwrite_existing)`. So I think both `cleanup` and  `overwrite_existing` can be a keyword argument.